### DISCUSSION TITLE
Find a block that is neither an image nor text, and get_text() method cannot extract

### DISCUSSION THREAD

**Discussion by 1339503169** [pymupdf/PyMuPDF#2587] [Q&A] [answered]
when i try to extract text from this file, i find something that i can choose it in WPS but can not be extract by get_text() method 

just use page.get_text() and 'BILL OF LADING NO' can not be found 

my pymupdf version is  1.18.8

and also i provide a picture that this something shows in WPS

i hope you guys will answer me soon 

thank you  

## Additional context (optional)
[2721523790.pdf](https://github.com/pymupdf/PyMuPDF/files/12287960/2721523790.pdf)
![feature1](https://github.com/pymupdf/PyMuPDF/assets/22074904/f6e99e67-c02a-4d97-af11-87a6810c6a50)
![feature2](https://github.com/pymupdf/PyMuPDF/assets/22074904/4bae8a1c-bdb6-422d-bef2-00af2363d00e)

**Comment by JorjMcKie** [comment]
On which page is that text "BILL OF LADING NO"? Don't see it anywhere.

**Reply by 1339503169** [reply]
BILL OF LADING NO is in the first page

**Comment by JorjMcKie** [comment]
Also, please what are your attached pictures supposed to tell me?

**Reply by 1339503169** [reply]
https://github.com/pymupdf/PyMuPDF/discussions/2587#discussioncomment-6681241
In this discussion, I gave a more detailed description.

**Comment by JorjMcKie** [comment]
Well, this obviously is no issue, but a Discussions topic.

**Comment by 1339503169** [comment]
![feature3](https://github.com/pymupdf/PyMuPDF/assets/22074904/a65140c4-73ba-4359-9c1d-76fc04f9fc1d)
BILL OF LADING NO is in the first page of the pdf which name is [2721523790.pdf](https://github.com/pymupdf/PyMuPDF/files/12287960/2721523790.pdf) and i marked it in red box

![feature4](https://github.com/pymupdf/PyMuPDF/assets/22074904/d838d42e-354a-4bf0-bfdf-fb9472213b71)
in this picture, you can see the 'B' in BILL OF LADING NO has been enlarged by WPS, which means there is a object represent this B, but by using fitz,  I could not identify the ”B“ by either page.get_text() or page.get_pixmap() , so what class in fitz that represent this 'B', can i  identify  it by using fitz?  this confuse me a lot

**Reply by JorjMcKie** [reply, marked as answer]
These things are vector graphics. Meaning they consist of drawing commands: lines, curves, rectangles.
For example, you can draw capital Letter "A" by these lines: `/-\` if you know how to do this.
PyMuPDF can extract these too via method `page.get_drawings()`. Please consult the documentation.
Vector graphics have the great advantage, that they remain smooth when the page image is zoomed in - raster images become pixelized.
